src/linker/Linker/LinkContext.cs

```diff
@@ -502,21 +502,6 @@ public void LogMessage (MessageContainer message)
 				message.Category == MessageCategory.Info) && !LogMessages)
 				return;
 
-			if (message.Category == MessageCategory.Warning &&
-				NoWarn.Contains ((uint) message.Code)) {
-				// This warning was turned off by --nowarn.
-				return;
-			}
-
-			// Note: message.Version is nullable. The comparison is false if it is null.
-			// Unversioned warnings are not controlled by WarnVersion.
-			// Error messages are guaranteed to only have a version if they were created for a warning due to warnaserror.
-			if ((message.Category == MessageCategory.Warning || message.Category == MessageCategory.Error) &&
-				message.Version > WarnVersion) {
-				// This warning was turned off by --warn.
-				return;
-			}
-
 			if (OutputWarningSuppressions && message.Category == MessageCategory.Warning && message.Origin?.MemberDefinition != null)
 				WarningSuppressionWriter.AddWarning (message.Code.Value, message.Origin?.MemberDefinition);
```

Comment:
We can also move this to CreateWarningMessage.

Reply:
I think that code is correct here. The warning should go to output file only when it's printed out not if it's constructed.